### PR TITLE
feat: Update EKS module version and add additional variables supported by the module

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?

- Update EKS module version and add additional variables supported by the module
- Add support for disabling cluster encryption
- Remove workflow scripts that have since moved to a `scripts/` directory

### Motivation

- Internal request

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: .github/workflows/iam-policy-generator.py [hunk 1]
```diff
@@ -1,35 +0,0 @@
-import json
```
Comment (bryantbiggs) [inline .github/workflows/iam-policy-generator.py:1]: This lives here https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/.github/scripts/iam-policy-generator.py

**Review by Zvikan — APPROVED** [review]

File: .github/workflows/plan-examples.py [hunk 2]
```diff
@@ -1,30 +0,0 @@
-import json
```
Comment (bryantbiggs) [inline .github/workflows/plan-examples.py:1]: this lives here https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/.github/scripts/plan-examples.py

File: main.tf [hunk 3]
```diff
@@ -12,77 +18,64 @@ module "kms" {
   tags                    = var.tags
 }
 
-# ---------------------------------------------------------------------------------------------------------------------
-# EKS CONTROL PLANE
-# ---------------------------------------------------------------------------------------------------------------------
 module "aws_eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "v18.26.6"
+  version = "v18.29.1"
```
Comment (bryantbiggs) [inline main.tf:23]: we don't want to go up to v18.30 if we can avoid it - 18.30 added default tags logic that is proving to be problematic and will be removed/reverted in v19.x

**Review by askulkarni2 — APPROVED** [review]
🙌🏽

File: main.tf [hunk 4]
```diff
@@ -1,8 +1,14 @@
-# ---------------------------------------------------------------------------------------------------------------------
-# CLUSTER KMS KEY
-# ---------------------------------------------------------------------------------------------------------------------
+locals {
+  cluster_encryption_config = length(var.cluster_encryption_config) > 0 ? [
```
Comment (muram) [inline main.tf:2]: Hi @bryantbiggs , I believe this conditional should evaluate if the `cluster_encryption_config` variable is not provided which means `== 0` not `> 0`. The default value is an empty list `[]`.

Comment (bryantbiggs) [reply main.tf:2]: ah yes,  you are correct. let me fix this

Comment (bryantbiggs) [reply main.tf:2]: PR https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1137

Thank you for catching this @muram

**Pull request merged**